src/Service/BrkService.php

```diff
@@ -304,22 +340,25 @@ public function mapZakelijkGerechtigden(array $snapshot): array
         foreach ($zakelijkGerechtigden as $zakelijkGerechtigde) {
             if ($zakelijkGerechtigde['parent'] !== '') {
                 $previousParent = $zakelijkGerechtigde['parent'];
+            } else if (isset($previousParent) === false) {
+                $zakelijkGerechtigde['parent'] = null;
             } else {
                 $zakelijkGerechtigde['parent'] = $previousParent;
             }
 
-            $object          = $this->handleRefObject($zgSchema, $zakelijkGerechtigde);
-            $onroerendeZaken = $this->cacheService->searchObjects(
-                '',
-                [
-                    'identificatie'    => $zakelijkGerechtigde['parent'],
-                    '_self.schema.ref' => 'https://brk.commonground.nu/schema/kadastraalOnroerendeZaak.schema.json',
-                ]
-            )['results'];
+            $object = $this->handleRefObject($zgSchema, $zakelijkGerechtigde);
+            if (isset($this->onroerendeZaken[$zakelijkGerechtigde['parent']]) === true) {
+                $onroerendeZaak = $this->onroerendeZaken[$zakelijkGerechtigde['parent']];
+
+                if (isset($zakelijkGerechtigde['hoofdsplitsing']) === true) {
+                    $this->addToCache($zakelijkGerechtigde['hoofdsplitsing'], $onroerendeZaak);
```

Comment:
Lot of nesting, maybe use 
if === false
continue;
Instead, to reduce nesting levels?